Source/WebCore/loader/cache/CachedResource.cpp

```diff
@@ -376,6 +380,22 @@ void CachedResource::finish()
         setStatus(Cached);
 }
 
+void CachedResource::setLoading(bool b)
+{
+    m_loading = b;
+    if (m_loading)
+        return;
+    auto callbacks = std::exchange(m_loadedCallbacks, { });
+    for (auto& callback : callbacks)
+        callback();
+}
+
+void CachedResource::whenLoaded(Function<void()>&& callback)
```

Comment:
It seems like this is called once and exactly once.  It could be a CompletionHandler.